### ISSUE TITLE
New Feature: Open Projects

### ISSUE THREAD

**Issue by bnosek** [CenterForOpenScience/osf.io#1736]
For next epic or beyond.

Setting projects as Open is one step beyond public.  With Open projects, OSF users can join the project themselves (as Read+Write).

For public projects, when the administrator sets them to open the public button becomes a "Join" button.  Pressing it either pops-up a modal to log-in/sign-up or adds the person to the contributor list.

Administrators can set the project to open when the make the project public and from the settings page.  They are warned that this means that any OSF user can join the project, edit the wiki, and add files.

**Comment by bnosek** [comment]
might make sense to add at the same time as adding "Request Access" feature - projects/components expose the title and description and users can request read access.

**Comment by fabianvf** [comment]
Interesting. Would this include a system to undo changes, similar to wikipedia, to help mitigate unhelpful or malicious users?

**Comment by bnosek** [comment]
Yes, it would need that over time.  But an incremental approach can just
add the feature and warn users about the consequences of turning it on.
Then, if there is use and interest, we could expand the accessibility of
version control for roll-back.

**Comment by nanchen47** [comment]
related to issue https://github.com/CenterForOpenScience/osf.io/issues/2288
